js/web/lib/wasm/jsep/webgpu/ops/common.ts

```diff
@@ -652,7 +656,7 @@ export const outputVariable =
     (name: string, type: number, shapeOrRank: number|readonly number[], components: 1|2|3|4 = 1): IndicesHelper =>
         createIndicesHelper(name, type, shapeOrRank, false, components);
 
-export type UniformsArrayType = Array<{name: string; type: string}>;
+export type UniformsArrayType = Array<{name: string; type: string; length?: number}>;
```

Comment:
Yes, add check at [1]

Reply:
can we make constraint on the "type" here to scalar type?

```suggestion
export type UniformDataElementType = 'u32'|'f32'|'i32';
export type UniformsArrayType = Array<{name: string; type: UniformDataElementType; length?: number}>;
```

since we already have a "length" property we don't have to support too complicated usages. with this we can simplify the later code to check `if (type.includes('vec')) {` and so on.